server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java

```diff
@@ -264,7 +278,7 @@ public void testOnlyBlocksOnConnectionsToNewNodes() throws Exception {
             assertConnectedExactlyToNodes(nodes1);
 
             nodeConnectionBlocks.clear();
-            connectionBarrier.await();
+            connectionBarrier.await(0, TimeUnit.SECONDS);
```

Comment:
Thinking more about this case again, it seems we could be unfortunate enough that the connecting thread has not yet reached the barrier, since we could first timeout after 1ms (which is still a long time, but in theory the connecting thread could be slowed by other activity) and then cancel the connect making future6.actionGet complete immediately.

So we should add 10 seconds timeout here. We can keep 0 timeout for the previous `await`

Reply:
Ok, I went round in circles on this a bit but I think you're right.